price-estimator/src/infallible_price_source.rs

```diff
@@ -0,0 +1,72 @@
+use core::{models::TokenId, token_info::TokenBaseInfo};
+use std::collections::HashMap;
+
+/// Roughly like `PriceSource` but is updated externally and cannot fail.
+#[derive(Debug, Default)]
+pub struct InfalliblePriceSource {
```

Comment:
I'm happy with this just being a struct for now and implementing a trait if it is needed later. Especially since it is fairly easy to construct one of these for testing.